lib/Bio/EnsEMBL/DataCheck/Checks/MySQLStorageEngine.pm

```diff
@@ -0,0 +1,51 @@
+=head1 LICENSE
+
+Copyright [2018] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::MySQLStorageEngine;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME        => 'MySQLStorageEngine',
+  DESCRIPTION => 'Check if database schema does not match the given MySQL storage engine',
+};
+
+sub tests {
+  my ($self) = @_;
+  my $helper = $self->dba->dbc->sql_helper;
+  my $database_name = $self->dba->dbc->dbname;
+  my $engine = 'MyISAM';
+  # The test should pass is a database schema match the given MySQL storage engine
+  # It should fail and report tables that are not on the given MySQL storage engine
+  my $desc = "All tables are using MySQL $engine storage engine";
```

Comment:
I suggest setting a $diag variable and passing it as the final parameter for is_rows_zero. This could be something like "Non-$engine table". For failures this would print out, for example, "Non-MyISAM table: object_xref" rather than the generic default, "Unexpected data: object_xref".

Reply:
Make sense, thanks.